interface/main/holidays/import_holidays.php

```diff
@@ -0,0 +1,197 @@
+        !empty($_POST['import_holidays'])       &&
+        !empty($_POST['sync'])
+        ){
+    echo "<p style='color:red'>" .
+        xlt('Operation Failed' );
+    "</p>\n";
+}
+?>
+<form method='post' action='import_holidays.php' enctype='multipart/form-data'
+      onsubmit='return top.restoreSession()'>
+
+        <p class='text'>
+        <table border='1' cellpadding='4'>
+            <tr bgcolor='#dddddd' class='dehead'>
+                <td align='center' colspan='2'>
+                    <?php echo xlt('CSV'); ?>
+                </td>
+            </tr>
+            <tr>
+                <td class='detail' nowrap>
+                    <?php echo htmlspecialchars(xl('CSV File')); ?>
+                    <input type="hidden" name="MAX_FILE_SIZE" value="350000000" />
+                </td>
+                <td class='detail' nowrap>
+                    <input type="file" name="form_file" size="40" />
+                </td>
+            </tr>
+            <tr>
+                <td class='detail' nowrap>
+                    <?php echo xlt('File on server (modification date)'); ?>
+                </td>
+                <td class='detail' nowrap>
+                    <?php
+                    if(!empty($csv_file_data)){?>
+
+                        <?php $path=explode("/",$holidays_controller->get_target_file());?>
+                        <?php $filename=$path[count($path)-1];?>
+                        <?php unset($path[count($path)-1]);?>
+
+                        <?php
+                        // build the path to the for the download request
+                        $phppagepath = "http://$_SERVER[HTTP_HOST]$_SERVER[REQUEST_URI]?filename=";?>
+                        <a href="#" onclick='window.open("import_holidays.php?download_file=1")'><?php echo text($csv_file_data['date']);?></a>
+                    <?php }else{
+                        echo htmlspecialchars(xl('File not found'));
+                    }?>
+                </td>
+            </tr>
+
+        <tr bgcolor='#dddddd'>
+                <td align='center' class='detail' colspan='2'>
+                    <input type='submit' name='bn_upload' value='<?php echo xla('Upload / Save') ?>' />
+                </td>
+            </tr>
+        </table>
+        </p>
+</form>
+        <table border='1'>
+
+        <tr >
+            <td >
+                <form method='post' action='import_holidays.php' onsubmit='return top.restoreSession()'>
+                    <input type='submit' name='import_holidays' value='<?php echo xla('Import holiday events') ?>'></br>
+
+                </form>
+            </td>
+
+            <td>
+                    <?php echo xla('CSV to calendar_external table'); ?></br>
+                <?php echo xla('If  the csv has been uploaded please click on "Import holiday events" button
+                    **NOTE: clicking on the button will remove all the existing rows in the calendar external table')?>
```

Comment:
I know this is ugly, but translations need to stay on line. line breaks will break the engine:
``````
<?php echo xla('If  the csv has been uploaded please click on "Import holiday events" button. NOTE that clicking on this button will remove all the existing rows in the calendar_external mysql table'): ?>